### DISCUSSION TITLE
Parallel Plotting & Queue

### DISCUSSION THREAD

**Discussion by Elerong** [Chia-Network/chia-blockchain#3192] [General]
Can you please adjust a function for Parallel plotting?

When a user selects Parallel plotting, select the number of plots that should work in parallel and how many plots would you like in a queue?

The issue begins when ME a user creates 6 parallel plots 85min delay there is no queue created (technically).
The application does not register those parallel plots as a part of Queue named "default"

When I create Queue of any size lets say 29 plots with a "default" name, the 1st plot of that queue starts automatically BUT I don't want it to.
Making total of 2 plots running at the same time.

Alternative solution would be stop the 2nd plot "Add Plot to Queue" from starting.
So once 7 plots are declared for parallel with 85 min delay.

Then I do again "Add Plot" followed by "Add Plot to Queue" then it just keeps the plots idle until any of those 7 plots finishes.

I wish I could just create 7 parallel plots with 85 delay and 29 plots after that which are waiting in a que.

**Comment by razamataz** [comment]
используй планировщик

**Comment by Elerong** [comment]
> используй планировщик
What do you mean? 
I am referring to Windows Client with the GUI

**Comment by razamataz** [comment]
используй файл CMD , командная строка, для очереди - n
используй Task Scheduler для запуска по времени.

**Comment by C0braChicken** [comment]
110% agree with Elerong's  post. The que and plot system just doesn't make any sense the way it's setup. It's like a big game as to how to get the right amount of plots going at the same time, due to the exact issue he describes. After you have any plots running, all queued plots should be added to the que, none of them should start until another is finished.

The weird thing is after the system has been running for half a day, the que system works the right way, and you can add as many ques as you like with no issue. But getting the first set of plots running in parallel with timings and queued working is a complete nightmare. due to queued plots starting before any of the parallel plots are finished

Also why is there an option to delete a plot, but it is just grayed out? So if the queued plots are misconfigured, I can't just delete the misconfigured ones, and add new queued plots, I have to kill the whole program, and restart everything. I bet I have lost over 20 plots trying to get the plotting system working correctly.

**Reply by C0braChicken** [reply]
LMAO.. after typing all of the post below, I translated what is in Russian above... and it says EXACTLY what my plan is.

So last night I had four plots 100% running from dual parallel plots, so I can have two in the under 31% range, and two above the 31% range running at the same time. Once all four were above 10% I then added 29 plots to the que, so the computer could just mine for a few days, however soon as I added them to the que, it automatically started one of them. I only have 1TB temp drive, so it's going to crash sometime due to running out of space, So now I will have to force CHIA to close, manually delete all the temp files, then attempt to get a good configuration going again. Since I have three plotters, I will have to do this to all three of my plotters. what a pain in the ass, and major lost plotting time.

My plan to test today. (need to double check the 150 minute part, with all notes I made last night)
Start a que with 29 plots, then set three alarms on my phone with the correct timings to start the next line.
Wait 150 minutes, start another que with 29 plots. If one doesn't auto start, then also add one parallel plot......
Wait 150 minutes, start another que with 29 plots. If one doesn't auto start, then also add one parallel plot......
Wait 150 minutes, start another que with 29 plots. If one doesn't auto start, then also add one parallel plot......
that will HOPEFULLY give me FOUR plots running at a time, with a que large enough to not have to mess with it for a long time. It will also hopefully keep running four plots, and not get confused as to how many plots should be running. 

I'm not sure if the flaw is with having dual parallel plots started? Then adding to the Que (always have default as name, I have never changed that). Maybe if I started one parallel plot with four in it, with the correct timings, then added the que that would work correctly, but I'm not going to waste my plotting time testing that. So today I will be testing out my above plan, and I'm 99% sure that should work.

This needs FIXED, it's a total pain in the ass... and making so my plotters are only getting a few plots a day due to having to restart the computers several times a day, so I can reconfigure the plots, and test out a different way to try to make it work. I bet I would have at least twice as many completed plots, if I wasn't having to force close and restart all the plots again, and again, and again... This is getting really old.

**Comment by razamataz** [comment]
You created a problem for yourself out of nowhere.
Use it https://github.com/Chia-Network/chia-blockchain/wiki/CLI-Commands-Reference
1. Add a directory to Windows Path: C:\Users\User\AppData\Local\chia-blockchain\app-1.1.2\resources\app.asar.unpacked\daemon
2. Create three CMD files in any folder
3. Example for you CDM file:  chia plots create -k 32 -r 4 -u 128 -t X:\ -d F:\ -n 29
4. Use the windows task scheduler to run CMD files on time, once. After about 2 hours each.
5. Enjoy

When you use this method you can safely close the Chia program, and even make plots when it is closed. 
In this variant, there will always be approximately one plotting process in the first phase, so I recommend using all processor threads, 8-16. (- r 16)

Sorry for my english

**Reply by C0braChicken** [reply]
I will try that later tonight, if I can figure out how to set it up, Sounds like a much better way to do this.

You English is fine, it is my native language, and you probably have better grammar than I do.

**Reply by engeldan** [reply]
подскажи пожалуйста, при запуске плота из-под командной строки (bat файл), закроется ли окно по завершению создания? Если нет, то как можно эту опцию включить? Спасибо

**Reply by razamataz** [reply]
закроется

как сделать точно не уверен, смотри справку CMD /?
возможно CMD /K
или Pause в конце файла

**Reply by engeldan** [reply]
мне как раз и надо, чтобы закрывалось. 
Мысль такая - делаю отдельный батник для старта плоттинга с ожиданием закрытия окна (start /w), как плот завершился - процесс создания стартует снова, таким образом, получается бесконечная очередь, пока сам не остановишь

**Reply by razamataz** [reply]
так добавь -n 100000 (или сколько тебе надо?) и будет очередь
-n[количество графиков]: количество графиков, которые будут созданы последовательно. Как только сюжет закончен, он будет перемещен в последнее место -dперед началом следующего сюжета в последовательности.
